File: src/me/krickl/memebotj/InternalCommands/HelpCommand.java
package me.krickl.memebotj.InternalCommands;

import me.krickl.memebotj.ChannelHandler;
import me.krickl.memebotj.CommandHandler;
import me.krickl.memebotj.UserHandler;

public class HelpCommand extends CommandHandler {

	public HelpCommand(String channel, String command, String dbprefix) {
		super(channel, command, dbprefix);
		this.setHelptext("Displays help for other commands");
	}

	@Override
	protected void commandScript(UserHandler sender, ChannelHandler channelHandler, String[] data) {
		try {
			int j = -1;
			if ((j = channelHandler.findCommand(data[0])) != -1) {
				channelHandler.sendMessage(channelHandler.getChannelCommands().get(j).getHelptext(),
						this.getChannelOrigin());

				return;
			}

			for (CommandHandler ch : channelHandler.getInternalCommands()) {
				if (ch.getCommand().equals(data[0])) {
					channelHandler.sendMessage(ch.getHelptext(), this.getChannelOrigin());

					return;
				}
			}

			channelHandler.sendMessage(channelHandler.getBuiltInStrings().get("HELP_NOT_FOUND"),
					this.getChannelOrigin());
		} catch (ArrayIndexOutOfBoundsException e) {
			channelHandler.sendMessage(
					channelHandler.getBuiltInStrings().get("HELP_SYNTAX").replace("{param1}", "!help <command>"),
					this.getChannelOrigin());
		}
	}

}

File: src/me/krickl/memebotj/InternalCommands/UserCommands/HelpCommand.scala
package me.krickl.memebotj.InternalCommands.UserCommands

import me.krickl.memebotj.{ChannelHandler, CommandHandler, Memebot, UserHandler}

//remove if not needed
import scala.collection.JavaConversions._

class HelpCommand(channel: String, command: String, dbprefix: String) extends CommandHandler(channel,
  command, dbprefix) {

  this.setHelptext("Displays help for other commands")

  override def commandScript(sender: UserHandler, channelHandler: ChannelHandler, data: Array[String]) {
    try {
      val j = channelHandler.findCommand(data(0))
      if (j != -1) {
        channelHandler.sendMessage(channelHandler.getChannelCommands.get(j).getHelptext, this.getChannelOrigin)
        return
      }
      for (ch <- channelHandler.getInternalCommands if ch.getCommand == data(0)) {
        if (ch.getHelptext == "null") {
          channelHandler.sendMessage(Memebot.formatText("HELP_NOT_FOUND", channelHandler, sender, this, true), this.getChannelOrigin)
        }
        else {
          channelHandler.sendMessage(ch.getHelptext, this.getChannelOrigin)
        }
        return
      }
      channelHandler.sendMessage(Memebot.formatText("HELP_NOT_FOUND", channelHandler, sender, this, true), this.getChannelOrigin)
    } catch {
      case e: ArrayIndexOutOfBoundsException => channelHandler.sendMessage(Memebot.formatText("HELP_SYNTAX", channelHandler, sender, this, true, Array("!help <command>")), this.getChannelOrigin)
    }
  }
}
